pallets/schemas/src/types.rs

```diff
@@ -1,13 +1,34 @@
 //! Types for the Schema Pallet
-use common_primitives::schema::{ModelType, PayloadLocation, SchemaSettings};
-use frame_support::traits::StorageVersion;
+use crate::{Config, Error};
+use common_primitives::schema::{ModelType, PayloadLocation, SchemaId, SchemaSettings};
+use frame_support::{ensure, pallet_prelude::ConstU32, traits::StorageVersion, BoundedVec};
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
+use sp_runtime::DispatchError;
 use sp_std::fmt::Debug;
+extern crate alloc;
+use alloc::string::{String, ToString};
+use sp_std::vec::Vec;
 
 /// Current storage version of the schemas pallet.
 pub const SCHEMA_STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
 
+/// The maximum size of schema name including all parts
+pub const SCHEMA_NAME_BYTES_MAX: u32 = 32; // Hard limit of 32 bytes
+/// A schema name following following structure NAMESPACE.DESCRIPTOR
+pub type SchemaNamePayload = BoundedVec<u8, ConstU32<SCHEMA_NAME_BYTES_MAX>>;
+/// The minimum size of a namespace in schema
+pub const NAMESPACE_MIN: usize = 3;
+/// The maximum size of a namespace in schema
+pub const NAMESPACE_MAX: u32 = 15;
+/// The minimum size of a schema descriptor
+pub const DESCRIPTOR_MIN: usize = 1;
+/// The maximum size of a schema descriptor
+pub const DESCRIPTOR_MAX: u32 = 16;
+/// separator character
+pub const SEPARATOR_CHAR: char = '.';
+const AUTO_ASSIGNED_NAMESPACE: &str = "auto_assigned";
```

Comment:
**Question:** what should this be?

currently we are using `auto_assign.<current_id>` for older extrinsics